### PR TITLE
drivers: display: fix a few bugs

### PR DESCRIPTION
**Pull request by gmarull**
Fix a few issues I've encountered in device drivers as a result of unrelated work (#42230).

### CODE REVIEW

File: drivers/display/ssd16xx.c [hunk 1]
```diff
@@ -700,6 +700,6 @@ static struct display_driver_api ssd16xx_driver_api = {
 
 
 DEVICE_DT_INST_DEFINE(0, ssd16xx_init, NULL,
-		    &ssd16xx_driver, NULL,
+		    &ssd16xx_driver, &ssd16xx_config,
```
Comment (jfischer-no) [inline drivers/display/ssd16xx.c:703]: No, do not change it, driver is not broken, regarding confusing configuration see https://github.com/zephyrproject-rtos/zephyr/pull/39708

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: drivers/display/ssd16xx.c (hunk 1 above)
Comment (gmarull) [reply drivers/display/ssd16xx.c]: I see, driver uses this weird pattern. Dropped for now (will eventually get done properly).

**Review by jfischer-no — APPROVED** [review]

File: drivers/display/gd7965.c [hunk 2]
```diff
@@ -429,7 +429,7 @@ static struct gd7965_data gd7965_driver = {
 	.config = &gd7965_config
 };
 
-static const struct gd7965_config gd7965_config {
+static const struct gd7965_config gd7965_config = {
```
Comment (MaureenHelm) [inline drivers/display/gd7965.c:432]: Do we ever build this driver in CI?

Comment (gmarull) [reply drivers/display/gd7965.c:432]: I guess not

Comment (JordanYates) [reply drivers/display/gd7965.c:432]: It's defined as part of a `build_all` test, https://github.com/zephyrproject-rtos/zephyr/blob/fab291779a4d857a09e2a21701f83b9f07c9c5db/tests/drivers/build_all/sensor/spi.dtsi#L83 but the test is missing the Kconfig options to actually compile it (`CONFIG_DISPLAY`, `CONFIG_GD7965`).

Comment (gmarull) [reply drivers/display/gd7965.c:432]: should actually not be in sensor/, right?

Comment (JordanYates) [reply drivers/display/gd7965.c:432]: true, but there isn't currently a `build_all/display`

**Review by nashif — APPROVED** [review]

**Pull request merged**